### ISSUE TITLE
Agent errors on 0.2.3: `Internal Server Error: invalid run state transition`

### ISSUE THREAD

**Issue by pat-s** [leg100/otf#673]
Updated from 0.1.18 to 0.2.3. Recreated agent tokens. 

Agents are picked up and can connect but throw the following errors in the logs:

```
│ 2023/12/15 10:43:45 INFO received job job.run_id=run-bpsejEK6uvAwPGC1 job.phase=plan job.status=allocated                                                                                                      │
│ 2023/12/15 10:43:45 ERROR starting job error="Internal Server Error: invalid run state transition" 
```

Runs do not start but are stuck in pending state, presumably due to the error?

![image](https://github.com/leg100/otf/assets/8430564/e5e5a69d-178b-4c2b-a8a5-967b85b3f85d)

**Comment by leg100** [comment]
Can you send over server logs? And set both to highest verbosity.

**Comment by pat-s** [comment]
With `-v 10` I see

```
2023/12/18 19:35:11 ERROR validating bearer token error="retrieving agent corresponding to ID found in http header: resource not found"
2023/12/18 19:35:11 ERROR starting plan error="invalid run state transition" id=run-PhVT96om3z0xZKh8
2023/12/18 19:35:11 ERROR validating bearer token error="retrieving agent corresponding to ID found in http header: resource not found"
2023/12/18 19:35:11 ERROR starting job error="invalid run state transition" spec=run-PhVT96om3z0xZKh8/plan agent.id=agent-KZGTdbg8vZOsokbz agent.server=true agent.status=idle agent.ip_address=172.17.0.2
2023/12/18 19:35:11 ERROR starting plan error="invalid run state transition" id=run-PhVT96om3z0xZKh8
```

Not much more unfortunately, the messages from above repeat themselves every few seconds.

**Comment by pat-s** [comment]
I meanwhile realized that I won't be able to execute a successful run anymore without solving this error - as the respective agent (in this case "Remote") won't be doing anything until this error is solved.

It's coming from specific runs which (most likely) were in a pending state during the upgrade. After the upgrade to 0.2.x, it is now in a state which is "unknown" and OTF can't deal with this.

I eventually went to delete these specific runs from the DB manually and afterward, runners started working normally again.
This applied to both the built-in "remote" runner and the "agent" runner.

I regularly see "pending" runs in the run history which either didn't start properly or were waiting for a manual approval. I assume that such runs which didn't "fully" finish might be candidates for the problem described in this issue (I didn't look at all runs in the end since there were too many - I deleted around ~ 20 runs manually from the DB).
